Framework/test/testCcdbDatabase.cxx

```diff
@@ -120,41 +128,71 @@ BOOST_AUTO_TEST_CASE(ccdb_store_for_future_tests)
-  TH1F* h1 = dynamic_cast<TH1F*>(mo->getObject());
-  BOOST_CHECK_NE(h1, nullptr);
-  BOOST_CHECK_EQUAL(h1->GetEntries(), 10000);
+  TH1F* h1 = new TH1F("to_be_kept", "asdf", 100, 0, 99);
+  h1->FillRandom("gaus", 12345);
+  shared_ptr<MonitorObject> mo1 = make_shared<MonitorObject>(h1, "task", "TST_KEEP");
+  mo1->addMetadata("Run", o2::quality_control::core::Version::GetQcVersion().getString());
+  shared_ptr<QualityObject> qo1 = make_shared<QualityObject>("check", vector{ string("input1"), string("input2") }, "TST_KEEP");
+  qo1->setQuality(Quality::Bad);
 
-  std::shared_ptr<TObject> obj2 = f.backend->retrieveTObject("qc/checks/TST/test-ccdb-check");
-  auto qo = dynamic_pointer_cast<QualityObject>(obj2);
-  BOOST_CHECK_NE(qo, nullptr);
-  BOOST_CHECK_EQUAL(qo->getQuality(), Quality::Bad);
+  f.backend->storeMO(mo1);
+  f.backend->storeQO(qo1);
 }
 
 BOOST_AUTO_TEST_CASE(ccdb_retrieve_mo, *utf::depends_on("ccdb_store"))
 {
   test_fixture f;
-  std::shared_ptr<MonitorObject> mo = f.backend->retrieveMO("qc/TST/my/task", "asdf/asdf");
-  BOOST_REQUIRE_NE(mo, nullptr);
-  TH1F* h1 = dynamic_cast<TH1F*>(mo->getObject());
-  BOOST_REQUIRE_NE(h1, nullptr);
-  BOOST_CHECK_EQUAL(h1->GetEntries(), 10000);
+  std::shared_ptr<MonitorObject> mo = f.backend->retrieveMO("qc/TST/my/task", "quarantine");
+  BOOST_CHECK_NE(mo, nullptr);
+  BOOST_CHECK_EQUAL(mo->getName(), "quarantine");
+}
 
-  std::shared_ptr<QualityObject> qo = f.backend->retrieveQO("qc/checks/TST/test-ccdb-check");
-  BOOST_REQUIRE_NE(qo, nullptr);
-  BOOST_CHECK_EQUAL(qo->getName(), "test-ccdb-check");
-  BOOST_REQUIRE_EQUAL(qo->getInputs().size(), 2);
-  BOOST_CHECK_EQUAL(qo->getInputs()[0], "input1");
-  BOOST_CHECK_EQUAL(qo->getInputs()[1], "input2");
-  BOOST_CHECK_EQUAL(qo->getDetectorName(), "TST");
+BOOST_AUTO_TEST_CASE(ccdb_retrieve_data_024)
+{
+  // test whether we can read data from version 0.24
+  test_fixture f;
+  shared_ptr<MonitorObject> mo = f.backend->retrieveMO("qc/TST_KEEP/task", "to_be_kept", 1585647354705);
+  BOOST_CHECK_NE(mo, nullptr);
+  BOOST_CHECK_EQUAL(mo->getName(), "to_be_kept");
+  BOOST_CHECK_EQUAL(dynamic_cast<TH1F*>(mo->getObject())->GetEntries(), 12345);
+
+  shared_ptr<QualityObject> qo = f.backend->retrieveQO("qc/checks/TST_KEEP/check", 1585647427642);
+  BOOST_CHECK_NE(qo, nullptr);
+  BOOST_CHECK_EQUAL(qo->getName(), "check");
+  BOOST_CHECK_EQUAL(qo->getQuality(), o2::quality_control::core::Quality::Bad);
+
+  auto jsonMO = f.backend->retrieveJson("qc/TST_KEEP/task/to_be_kept", 1585647354705);
+  BOOST_CHECK(!jsonMO.empty());
+
+  jsonMO = f.backend->retrieveMOJson("qc/TST_KEEP/task", "to_be_kept", 1585647354705);
+  BOOST_CHECK(!jsonMO.empty());
+
+  auto jsonQO = f.backend->retrieveJson("qc/checks/TST_KEEP/check", 1585647427642);
+  BOOST_CHECK(!jsonQO.empty());
+
+  jsonQO = f.backend->retrieveQOJson("qc/checks/TST_KEEP/check", 1585647427642);
+  BOOST_CHECK(!jsonQO.empty());
 }
 
+//BOOST_AUTO_TEST_CASE(ccdb_retrieve_data_025)
```

Comment:
is this still needed?

Reply:
yes. We will have to add it when updating the format in the future.